### PR TITLE
Add doc file for using external VISA configuration utility

### LINKED ISSUE #718: [COM] Communication issue with ARSL instruments using NI-VISA, configuration settings
NI-VISA comes with configuration utilities (MAX on Windows, visaconf on UNIX) for instrument communications settings.  These setting will not be used in pyVISA unless access_mode=constants.VI_LOAD_CONFIG is passed to open_resource().

`instr = rm.open_resource('ASRL18::INSTR', access_mode=constants.VI_LOAD_CONFIG)`

This NI-VISA behavior is consistent with the VISA library spec VPP-4.3 rev. 7.2. 
Keysight and R&S VISA pass the configured settings even if access_mode!=constants.VI_LOAD_CONFIG.

**This issue is for informational purposes only and can be closed.**

NI-VISA info:
C:\WINDOWS\system32\nivisa64.dll
        Comments: 2022/07/01 17:45:56, nivisa64/win64U/x64/msvc-14.0/release    
        InternalName: NIVISA_CORE 22.5.0f60
        ProductName: NIVISA_CORE
        CompanyName: National Instruments Corporation
        LegalCopyright: Copyright © 2000-2022 National Instruments Corporation. 
All Rights Reserved.
        ProductVersion: 22.5.0f60
        FileDescription: NI-VISA Library
        LegalTrademarks: None
        PrivateBuild: None
        FileVersion: 22.5.0f60
        OriginalFilename: nivisa64.dll
        SpecialBuild: NI-Spy

### PR DESCRIPTION
**Pull request by dphaas**
1. Added doc file for using external VISA configuration utility
2. Update broken links in existing document.

- [ ] Closes #718
- [ ] Executed ``black . && isort -c . && flake8`` with no errors
- [ ] The change is fully covered by automated unit tests
- [ ] Documented in docs/ as appropriate
- [ ] Added an entry to the CHANGES file

### CODE REVIEW

File: docs/source/faq/faq.rst [hunk 1]
```diff
@@ -206,34 +206,31 @@ Where can I get more information about VISA?
...
-       http://digital.ni.com/manuals.nsf/websearch/266526277DFF74F786256ADC0065C50C
+  - `VISA/IVI/SCPI specifications`_ (scroll down to find VISA)
+
+* The very good VISA manuals from NI:
 
+  - `NI-VISA Download`_
+  - `NI-VISA help file`_ in HTML (download in PDF, upper right corner)
+  - `R&S-VISA Download`_
+  - `R&S Basics of Instrument Remote Control`_
+  - `Keysight-VISA Download`_
```
Comment (MatthieuDartiailh) [inline docs/source/faq/faq.rst:217]: Since we also have a page dedicated to getting NI-VISA, it would make sense to move the download links there and maybe rework this a bit to be less NI centered.

File: docs/source/introduction/ext_cfg_util.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
+The purpose of the VISA external configuration utility
+==============================================================================
```
Comment (MatthieuDartiailh) [inline docs/source/introduction/ext_cfg_util.rst:2]: Please do not extend the underline beyond the title

File: docs/source/introduction/ext_cfg_util.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+Managing system VISA resources -
+------------------------------------------------------------------------------
+The three main VISA distributions all come with a utility to manage VISA
+resource configurations.  This utility is refered to as the external
+configuration utility.  The VISA external configuration utility is used to
+manage instrument attribute used to configure buses, instruments, and aliases.
+This configuration in turn can be used by pyVISA to manage instrument
```
Comment (MatthieuDartiailh) [inline docs/source/introduction/ext_cfg_util.rst:10]:
```suggestion
This configuration in turn can be used by PyVISA to manage instrument
```

File: docs/source/introduction/ext_cfg_util.rst [hunk 4]
```diff
@@ -0,0 +1,73 @@
+The purpose of the VISA external configuration utility
+==============================================================================
+
+Managing system VISA resources -
+------------------------------------------------------------------------------
+The three main VISA distributions all come with a utility to manage VISA
```
Comment (MatthieuDartiailh) [inline docs/source/introduction/ext_cfg_util.rst:6]: NI, R&S, Keysight I assume. Does Tek miss one ?

File: docs/source/introduction/ext_cfg_util.rst [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+.. note::
+
+    Not all VISA implementations handle settings configured in the external
+    utility the same.  For instance, in NI-VISA, if a serial port is configured
+    for baudrate = 115000, the default value will still be configured as 9600.
+    To pass the VISA configured value to a new instrument instance, access mode
+    must be set to constants.VI_LOAD_CONFIG (see below), otherwise, the default
+    value of 9600 will be passed to the instrument instance.  R&S-VISA doesn't
+    work this way.  (I was not able to load Keysight VISA to test it's
+    functionality at the time of this writing.)
```
Comment (MatthieuDartiailh) [inline docs/source/introduction/ext_cfg_util.rst:22]: Since I won't have the bandwidth to test, this will remain, so maybe rephrase in a less personal manner.

File: docs/source/introduction/ext_cfg_util.rst [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+     
+	open_resource(instr, access_mode=constants.VI_LOAD_CONFIG)
+
+    The important thing to remember is that most values set in the VISA
+    external     configuration utility are passed to pyVISA at instantiation of
```
Comment (MatthieuDartiailh) [inline docs/source/introduction/ext_cfg_util.rst:27]:
```suggestion
    external configuration utility are passed to pyVISA at instantiation of
```

Comment (MatthieuDartiailh) [inline docs/source/introduction/ext_cfg_util.rst:27]:
```suggestion
    external     configuration utility are passed to PyVISA at instantiation of
```

File: docs/source/introduction/ext_cfg_util.rst [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+compatible instrument with the same alias using the external configuration
+utility.
+
+Aliases provide a more robust way to refer to instruments for applications that
+must be propagated across multiple systems.  It also makes subsituting
+alternate instruments easier when required for maintenance or calibration.
+
+Find instruments and test communications -
+------------------------------------------------------------------------------
+Are instrument communications problems being caused by an instrument, the
```
Comment (MatthieuDartiailh) [inline docs/source/introduction/ext_cfg_util.rst:57]: Missing blank line

File: docs/source/introduction/index.rst [hunk 8]
```diff
@@ -14,6 +14,7 @@ communications issues.
     getting
     configuring
     communication
+    ext_cfg_util
```
Comment (MatthieuDartiailh) [inline docs/source/introduction/index.rst:17]: While valuable this is specific to the IVI backend so I would move it later (maybe last), or in the FAQ.

File: docs/source/introduction/ext_cfg_util.rst [hunk 9]
```diff
@@ -0,0 +1,73 @@
+The purpose of the VISA external configuration utility
```
Comment (MatthieuDartiailh) [inline docs/source/introduction/ext_cfg_util.rst:1]:
```suggestion
VISA external configuration utility for the IVI backend
```

File: docs/source/introduction/ext_cfg_util.rst [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+    The important thing to remember is that most values set in the VISA
+    external     configuration utility are passed to pyVISA at instantiation of
+    an instrument by default.  NI-VISA serial ports are an exception that
+    require:
+
+    "open_resource(instr, access_mode=constants.VI_LOAD_CONFIG)".
+
+Set VISA instrument alias -
+------------------------------------------------------------------------------
+A VISA instrument alias is one of the attributes that can be managed by the
```
Comment (MatthieuDartiailh) [inline docs/source/introduction/ext_cfg_util.rst:35]: Blank line

File: docs/source/introduction/ext_cfg_util.rst [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+external configuration utility. An alias is another way to refer to an
+instrument.  pyVISA can open an instrument using either the resource name or
```
Comment (MatthieuDartiailh) [inline docs/source/introduction/ext_cfg_util.rst:37]:
```suggestion
instrument.  PyVISA can open an instrument using either the resource name or
```

File: docs/source/introduction/ext_cfg_util.rst [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+the alias. Resource names are of the form 'GPIB0::3::INSTR', 'ASRL3::INSTR',
+'USB0::0x0699::0x0401::C021095::INSTR'.  What can be observed from these
+examples is that resource names are very hardware dependent.  All three of
+these resource names could refer to a DMM.  If the pyVISA app is coded with the
```
Comment (MatthieuDartiailh) [inline docs/source/introduction/ext_cfg_util.rst:41]:
```suggestion
these resource names could refer to a DMM.  If the PyVISA app is coded with the
```

**Review by MatthieuDartiailh — CHANGES_REQUESTED** [review]

File: docs/source/introduction/ext_cfg_util.rst (hunk 4 above)
Comment (dphaas) [reply docs/source/introduction/ext_cfg_util.rst:6]: NI, R&S, Keysight are "Sponsor" members of IVI foundation, but I see Tek and Keithley (a Tek company) are "General" members. `https://www.ivifoundation.org/membership/current_members.aspx`  I initially got the impression that Tek had abandoned providing/supporting TekVISA distributions.  After a more thorough search on the Tek website I did find: `https://www.tek.com/en/support/software/driver/tekvisa-connectivity-software-v420`

I personally think the fewer VISA distributions PyVISA mentions the easier it is to support users.  I would recommend omitting mention of TekVISA unless someone from Tek asks for it to be added.

Please let me know if you would like me to add TekVISA.

File: docs/source/introduction/ext_cfg_util.rst (hunk 2 above)
Comment (dphaas) [reply docs/source/introduction/ext_cfg_util.rst:2]: OK, fixed

File: docs/source/introduction/ext_cfg_util.rst (hunk 3 above)
Comment (dphaas) [reply docs/source/introduction/ext_cfg_util.rst:10]: OK, fixed.

File: docs/source/introduction/ext_cfg_util.rst (hunk 9 above)
Comment (dphaas) [reply docs/source/introduction/ext_cfg_util.rst:1]: OK, fixed.

File: docs/source/introduction/ext_cfg_util.rst (hunk 11 above)
Comment (dphaas) [reply docs/source/introduction/ext_cfg_util.rst:37]: OK, fixed.

File: docs/source/introduction/ext_cfg_util.rst (hunk 12 above)
Comment (dphaas) [reply docs/source/introduction/ext_cfg_util.rst:41]: OK, fixed.

File: docs/source/faq/faq.rst (hunk 1 above)
Comment (dphaas) [reply docs/source/faq/faq.rst:217]: I absolutely agree this should be less NI-VISA centric.  Let me give it a shot.

File: docs/source/introduction/index.rst (hunk 8 above)
Comment (dphaas) [reply docs/source/introduction/index.rst:17]: OK, moved to end.

File: docs/source/faq/faq.rst [hunk 13]
```diff
@@ -206,34 +206,31 @@ Where can I get more information about VISA?
...
-.. _`VISA specification for textual languages`:
-       http://www.ivifoundation.org/Downloads/Class%20Specifications/vpp432.doc
-.. _`National Instruments's VISA`: http://ni.com/visa/
-.. _`NI-VISA Programmer Reference Manual`:
-       http://digital.ni.com/manuals.nsf/websearch/87E52268CF9ACCEE86256D0F006E860D
-.. _`NI-VISA help file`:
-       http://digital.ni.com/manuals.nsf/websearch/21992F3750B967ED86256F47007B00B3
-.. _`NI-VISA User Manual`:
-       http://digital.ni.com/manuals.nsf/websearch/266526277DFF74F786256ADC0065C50C
+  - `VISA/IVI/SCPI specifications`_ (scroll down to find VISA)
```
[the remark this answers is not in the document]
Comment (dphaas) [reply docs/source/faq/faq.rst:209]: Added.

**Comment by MatthieuDartiailh** [comment]
Let me know when to review again

**Comment by dphaas** [comment]
@MatthieuDartiailh just got the checks to pass.  Ready for you to review now.

**Comment by MatthieuDartiailh** [comment]
I have not forgotten you, but I have a lot on my plate.I will try to complete my review today

File: docs/source/faq/visa_distributions.rst [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+no source code changes would be required.  One would simply add a new
+compatible instrument with the same alias using the external configuration
+utility.
+
+Aliases provide a more robust way to refer to instruments for applications that
+must be propagated across multiple systems.  It also makes subsituting
+alternate instruments easier when required for maintenance or calibration.
+
+Find instruments and test communications -
+------------------------------------------------------------------------------
```
Comment (MatthieuDartiailh) [inline docs/source/faq/visa_distributions.rst:82]: Underline is too long

File: docs/source/faq/visa_distributions.rst [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+multiple systems the names will have to be changed in the app source code.
+Alternately, the alias 'DMM' could be assigned to any of these instruments
+using the external configuration utility.  If the source code uses the alias,
+no source code changes would be required.  One would simply add a new
+compatible instrument with the same alias using the external configuration
+utility.
+
+Aliases provide a more robust way to refer to instruments for applications that
+must be propagated across multiple systems.  It also makes subsituting
+alternate instruments easier when required for maintenance or calibration.
```
Comment (MatthieuDartiailh) [inline docs/source/faq/visa_distributions.rst:79]: I am personally a bit bothered by this paragraph. I have developed and deployed several application and I have always preferred to provide this level of abstraction inside the app/in configuration files to avoid the need to rely on an external tool.  So I think I would like to see this toned down a bit.

**Review by MatthieuDartiailh — COMMENTED** [review]
I have a couple of comments, since they are a bit subjective let me if you want me to address them.
